torch/lib/THCUNN/generic/AbsCriterion.cu

```diff
@@ -35,25 +46,39 @@ void THNN_(AbsCriterion_updateGradInput)(
            THCState *state,
            THCTensor *input,
            THCTensor *target,
+           THCTensor *gradOutput,
            THCTensor *gradInput,
-           bool sizeAverage)
+           bool sizeAverage,
+           bool reduce)
 {
   THCUNN_check_nElement(state, input, target);
-  THCUNN_assertSameGPU(state, 3, input, target, gradInput);
+  THCUNN_assertSameGPU(state, 4, input, target, gradOutput, gradInput);
+
+  THCTensor_(resizeAs)(state, gradInput, input);
+
+  if (!reduce) {
+    THCUNN_check_nElement(state, gradOutput, input);
+    THC_pointwiseApply3(state, input, target, gradInput,
+                        abs_updateGradInput_no_reduce_functor<real>());
+    THC_pointwiseApply2(state, gradOutput, gradInput,
+                        multiply_and_store_functor<real>());
```

Comment:
I think this is just `THCTensor_(cmul)` and there's no need for an extra functor